src/plugins/intel_cpu/src/nodes/kernels/scaled_attn/executor_pa.cpp

```diff
@@ -1162,33 +1163,58 @@ struct MHAHelper {
         // aligned to cache line (64bytes=16*sizeof(float)) to avoid false sharing
         _weight_bhl.resize<float>({B, _H, q_len, rnd_up(max_context_len, std::max(_block_size, size_t{16}))});
 
-        parallel_for3d_dynamic(B, kv_len_in_blocks, _Hk, [&](size_t b, size_t pk_in_blocks, size_t hk) {
+        // for small batches dynamic scheduler has notable overhead
+        bool prefer_static_loop;
+        bool loop_hk = B * kv_len_in_blocks * _Hk > 2 * _nthr;
+        if (B <= 32) {
+            prefer_static_loop = true;
+            // small batch and all batch size is same(like SDPA case)
+            auto kv_len = past_lens.ptr<int32_t>()[0];
+            for (size_t b = 1; b < B; b++) {
+                if (past_lens.ptr<int32_t>()[b] != kv_len)
+                    prefer_static_loop = false;
+            }
+        } else {
+            // bigger batch, probably it's vllm path, skip the test to save the cost
+            prefer_static_loop = false;
+        }
+        auto loop_qk = [&](size_t b, size_t pk_in_blocks, size_t hx) {
             auto context_len = static_cast<size_t>(past_lens.ptr<int32_t>()[b]) + 1;
+            size_t hk, hq_beg, hq_end;
```

Comment:
The setting of `hq_beg` and `hg_end` are the same between `loop_qk` and 'loop_wk', please consider combine them together to a common place

Reply:
Done.